pkg/vendir/fetch/helmchart/oci_source.go

```diff
@@ -176,21 +176,28 @@ func (t *OCISource) addAuthArgs(args []string) ([]string, io.Reader, error) {
 			return nil, nil, err
 		}
 
-		secret, err = secret.ToBasicAuthSecret()
+		secrets, err := secret.ToRegistryAuthSecrets()
 		if err != nil {
 			return nil, nil, err
 		}
 
-		for name, val := range secret.Data {
-			switch name {
-			case ctlconf.SecretK8sCorev1BasicAuthUsernameKey:
-				authArgs = append(authArgs, []string{"--username", string(val)}...)
-			case ctlconf.SecretK8sCorev1BasicAuthPasswordKey:
-				authArgs = append(authArgs, []string{"--password-stdin"}...)
-				passwordStdin = strings.NewReader(string(val))
-			default:
-				return nil, nil, fmt.Errorf("Unknown secret field '%s' in secret '%s'", name, secret.Metadata.Name)
+		for _, secret := range secrets {
+			for name, val := range secret.Data {
+				switch name {
+				case ctlconf.SecretRegistryHostnameKey:
+					// do nothing for now
+					// TODO match secret by hostname?
+				case ctlconf.SecretK8sCorev1BasicAuthUsernameKey:
+					authArgs = append(authArgs, []string{"--username", string(val)}...)
+				case ctlconf.SecretK8sCorev1BasicAuthPasswordKey:
+					authArgs = append(authArgs, []string{"--password-stdin"}...)
+					passwordStdin = strings.NewReader(string(val))
+				default:
+					return nil, nil, fmt.Errorf("Unknown secret field '%s' in secret '%s'", name, secret.Metadata.Name)
+				}
 			}
+			// Only single set of credentials is supported
```

Comment:
this is a little opaque -- is the behavior that we're looping through the secrets until we find the first one that looks like it has auth credentials?   might be good to advertise that somehow,  whether in docstrings, user-facing documentation -  not sure who needs to know,  but seems like a potential stumbling block for a naive operator.

Reply:
👍 docs update coming to carvel repo.

Reply:
simplified logic for now to only allow 0/1 secrets: https://github.com/vmware-tanzu/carvel-vendir/commit/bdb4ba833bbb40f34d3c70635d8ae516059d9f56